src/libpmemfile-posix/file.c

```diff
@@ -986,21 +986,21 @@ vinode_exchange(PMEMfilepool *pfp,
 		 * If both are regular files or have the same parent, then
 		 * we don't have to do anything.
 		 */
-		if ((src_is_dir || dst_is_dir) && src_parent != dst_parent) {
+		if ((src_is_dir || dst_is_dir) && src->vinode != dst->vinode) {
```

Comment:
BTW, since vinode in path_info always points some parent directory:
```
/* path resolution information */
struct pmemfile_path_info {
        /*
         * Vinode of the last reachable component in the path, except for
         * the last part.
         */
        struct pmemfile_vinode *vinode;
```
Could it be called `parent_vinode`, or something similar?